app/assets/javascripts/surveys.js

```diff
@@ -53,6 +53,9 @@
 
       var $surveyLink = $('#take-survey');
       var surveyUrl = survey.url;
+      if (typeof surveyUrl === "function") {
+        surveyUrl = surveyUrl();
+      }
```

Comment:
The `javascript:void;` href is a placeholder, it's replaced by the value of `surveyUrl` (maybe there's a better way?).